cpp/rrtmgp_const.h

```diff
@@ -26,7 +26,11 @@ template <typename T, typename Device=DefaultDevice>
 using FOView = Kokkos::Experimental::OffsetView<T, Kokkos::LayoutLeft, Device>;
 
 template <int Rank, typename ExecutionSpace=Kokkos::DefaultExecutionSpace>
-using MDRangeP = Kokkos::MDRangePolicy<ExecutionSpace, Kokkos::Rank<Rank> >;//, Kokkos::Iterate::Right, Kokkos::Iterate::Left> >;
+#ifdef KOKKOS_ENABLE_CUDA
+using MDRangeP = Kokkos::MDRangePolicy<ExecutionSpace, Kokkos::Rank<Rank, Kokkos::Iterate::Left, Kokkos::Iterate::Right> >;
```

Comment:
Since rrmgp doesn't really use tiles, I expected the last template param to have no impact. That is, `Rank<Rank,Left,Left>` or even `Rank<Rank,Left>` should have given the same perf on average.

That said, the default outer dir for Cuda _is_ `Left`, so the fact that you say Left-Right performs better than leaving blank means that the tiling machinery is not exactly as I thought...

Reply:
@bartgol , yeah, it's very confusing. Fiddling with this MDRangeP type can have a big impact on performance on both CPU and GPU and the faster choice is always the opposite of what I thought it should be. I may just ask their slack channel about this

Reply:
Yeah, I was gonna suggest precisely that.

Reply:
YAKL multi-dimensional `parallel_for` works a little bit differently than Kokkos `MDRangePolicy`. Regardless of YAKL style (Fortran or C), the rightmost index given to a YAKL_LAMBDA is always the fastest varying. Without reordering indices, it doesn't surprise me that Left-Right performs better on GPU. Kokkos MDRangePolicy always does tiling, if you don't specify their size you just get default values. However, the defaults do not depend on the chosen iteration order, and are pretty bad for Left-Right. So I suggest to try different tile sizes to further improve performance. Alternatively, you could try reordering indices in Kokkos lambdas.

Reply:
Actually, I was wrong about the default tile sizes not depending on the iteration order. Sorry, not sure why I though that. Trying different tile sizes might still be beneficial.